primitives/core/src/lib.rs

```diff
@@ -331,6 +331,11 @@ pub fn to_substrate_wasm_fn_return_value(value: &impl Encode) -> u64 {
 	res
 }
 
+/// The void type - it cannot exist.
+// Oh rust, you crack me up...
+#[derive(Clone, Decode, Encode, Eq, PartialEq, RuntimeDebug)]
+pub enum Void {}
```

Comment:
I see this is used in this PR, can you explain a bit more about why this is introduced vs. using `()` as is often done?

I see `Void` could be used as a more explicit replacement for `()`.

Reply:
`KeyOwnerProofSystem` is implemented for `()` with the `KeyOwnerProof` type set to `Void` since no key owner proof can be constructed. I moved `Void` from `frame/support` to `primitives/core` so that it can be used in `primitives/session` for implementing the utility traits `GetValidatorCount` and `GetSessionNumber` on it.